src/interp/interp-wasi.cc

```diff
@@ -53,6 +53,49 @@ _Static_assert(_Alignof(__wasi_size_t) == 4, "witx calculated align");
 _Static_assert(sizeof(__wasi_ptr_t) == 4, "witx calculated size");
 _Static_assert(_Alignof(__wasi_ptr_t) == 4, "witx calculated align");
 
+typedef struct __wasi_prestat_dir_t {
+  __wasi_size_t pr_name_len;
+} __wasi_prestat_dir_t;
+
+typedef uint8_t __wasi_preopentype_t;
+typedef uint64_t __wasi_rights_t;
+typedef uint16_t __wasi_fdflags_t;
+typedef uint8_t __wasi_filetype_t;
+typedef uint16_t __wasi_oflags_t;
+typedef uint32_t __wasi_lookupflags_t;
+typedef uint32_t __wasi_fd_t;
+typedef uint64_t __wasi_timestamp_t;
+typedef uint8_t __wasi_whence_t;
+typedef int64_t __wasi_filedelta_t;
+typedef uint64_t __wasi_filesize_t;
+
+typedef union __wasi_prestat_u_t {
+  __wasi_prestat_dir_t dir;
+} __wasi_prestat_u_t;
+
+struct __wasi_prestat_t {
+  __wasi_preopentype_t tag;
+  __wasi_prestat_u_t u;
+};
+
+typedef struct __wasi_fdstat_t {
+  __wasi_filetype_t fs_filetype;
+  __wasi_fdflags_t fs_flags;
+  __wasi_rights_t fs_rights_base;
+  __wasi_rights_t fs_rights_inheriting;
+} __wasi_fdstat_t;
+
+_Static_assert(sizeof(__wasi_fdstat_t) == 24, "witx calculated size");
```

Comment:
nit: nicer to use C++ `static_assert`

Reply:
I'm going to leave as is because these copied directly of wasi-libc api.h and I plan on automating all of this soon.   Will clarify in the comment.